InternalDocs/tier2.md

```diff
@@ -0,0 +1,67 @@
+# The Tier 2 Interpreter
+
+The [basic interpreter](interpreter.md), also referred to as the `tier 1`
+interpreter, consists of a main loop that executes the bytecode instructions
+generated by the [bytecode compiler](compiler.md) and their
+[specializations](adaptive.md). Runtime optimization in tier 1 can only be
+done for one instruction at a time. The `tier 2` interpreter is based on a
+mechanism to replace an entire sequence of bytecode instructions, and this
+enables optimizations that span multiple instructions.
+
+## The Optimizer and Executors
+
+The program begins running in tier 1, until a `JUMP_BACKWARD` instruction
+determines that it is `hot` because the counter in its
+[inline cache](interpreter.md#inline-cache-entries) indicates that is
+executed more than some threshold number of times. It then calls the
+function `_PyOptimizer_Optimize()` in
+[`Python/optimizer.c`](../Python/optimizer.c), passing it the current
+[frame](frames.md) and instruction pointer. `_PyOptimizer_Optimize()`
+constructs an object of type
+[`_PyExecutorObject`](Include/internal/pycore_optimizer.h) which implements
+an optimized version of the instruction trace beginning at this jump.
+
+The optimizer determines where the trace ends, and the executor is set up
+to either return to `tier 1` and resume execution, or transfer control
+to another executor (see `_PyExitData` in Include/internal/pycore_optimizer.h).
+
+The executor is stored on the [`code object`](code_objects.md) of the frame,
+in the `co_executors` field which is an array of executors. The start
+instruction of the trace (the `JUMP_BACKWARD`) is replaced by an
+`ENTER_EXECUTOR` instruction whose `oparg` is equal to the index of the
+executor in `co_executors`.
+
+## The uop optimizer
+
+The optimizer that `_PyOptimizer_Optimize()` runs is configurable
+via the `_Py_SetTier2Optimizer()` function (this is used in test
+via `_testinternalcapi.set_optimizer()`.)
```

Comment:
Is this the API we plan to remove?  https://github.com/python/cpython/issues/126599

Reply:
The PR that removes it should update the docs.